server/db/schema/hostels.schema.ts

```diff
@@ -0,0 +1,48 @@
+import { relations } from 'drizzle-orm';
+import { pgTable, serial, text, timestamp, varchar } from 'drizzle-orm/pg-core';
+
+import { faculty, staff } from '.';
+
+export const hostels = pgTable('hostels', {
+  id: serial('id').primaryKey(),
+  address: text('address').notNull(),
+  name: varchar('name', { length: 256 }).notNull(),
+  urlName: varchar('url_name', { length: 128 }).notNull(),
+  email: varchar('email', { length: 256 }).notNull(),
+  telephone: varchar('telephone', { length: 13 }).notNull(),
+  alternateTelephone: varchar('alternate_telephone', { length: 13 }),
+  type: varchar('type', {
+    enum: ['boys', 'girls'],
+  }).notNull(),
+  createdAt: timestamp('created_at').defaultNow().notNull(),
+  updatedAt: timestamp('updated_at')
+    .$onUpdate(() => new Date())
+    .notNull(),
+});
+
+export const hostelsRelations = relations(hostels, ({ many }) => ({
+  staff: many(hostelStaff),
+  wardens: many(faculty, { relationName: 'wardens' }),
+  deputyWardens: many(faculty, { relationName: 'deputy-wardens' }),
+}));
+
+export const hostelStaff = pgTable('hostel_posts', {
+  hostelId: serial('hostel_id')
+    .references(() => hostels.id)
+    .notNull(),
+  staffId: serial('staff_id')
+    .references(() => staff.id)
+    .notNull(),
+  post: varchar('post', { enum: ['warden', 'deputy-warden'] }).notNull(),
```

Comment:
Post won't be `'warden' | 'deputy-warden'` anymore. Please see other staff roles and put those here